core/MatomoInstaller.php

```diff
@@ -0,0 +1,416 @@
+<?php
+
+namespace Piwik;
+
+use Piwik\Access;
+use Piwik\Common;
+use Piwik\Config;
+use Piwik\Container\StaticContainer;
+use Piwik\Db;
+use Piwik\Db\Adapter;
+use Piwik\DbHelper;
+use Piwik\Plugins\SitesManager\API as APISitesManager;
+use Piwik\Plugins\UsersManager\API as APIUsersManager;
+use Piwik\Plugins\UsersManager\NewsletterSignup;
+use Piwik\ProxyHeaders;
+use Piwik\Updater;
+use Piwik\Url;
+use Piwik\Version;
+use Zend_Db_Adapter_Exception;
+
+//Warning: - Don't know how plugin are installed (not done by installer
+//           actually). Some magics when I created installer.
+//
+// require : https://github.com/matomo-org/matomo/pull/15691
+
+/*
+ * Manage the Matomo server installation. Contain
+ * all basic function to setup database, user, and
+ * site to track.
+ *
+ * (?Could be a parent class to create custom
+ * installation process?)
+*/
+
+class MatomoInstaller{
+    /*
+     * Perform the entire installation of Matomo.
+     * Use user-defined settings.
+     *
+     * @settings is expected to be an array with following
+     * variables.
+     *
+     *  - dbname
+     *  - port (optionnal)
+     *  - dbusername
+     *  - dbpassword
+     *  - dbhost (optionnal?)
+     *  - tables_prefix (optionnal)
+     *  - adapter (optionnal)
+     *  - type (optionnal)
+     *  - adminusername
+     *  - adminpassword
+     *  - email
+     *  - subscribe_newsletter_piwikorg (optionnal)
+     *  - subscribe_newsletter_professionalservices (optionnal)
+     *  - name
+     *  - url
+     *  - ecommerce : (optionnal)
+    */
+    public static function headlessInstall($settings)
```

Comment:
btw ideally we would put these things maybe in a command so someone can specifically trigger the headless installation? We could create a command class and then add this command to https://github.com/matomo-org/matomo/blob/4.x-dev/core/Console.php#L270 so it's available even before any Matomo is installed . Or alternatively such a command be defined in a plugin and then we'd need to configure that plugin name in `[General]always_load_commands_from_plugin` setting.